src/tests/unittests/instrument_drivers/virtualAwg/awgs/test_ZurichInstrumentsHDAWG8.py

```diff
@@ -71,14 +71,14 @@ def test_retrieve_sampling_rate(self):
 
     def test_update_gain(self):
         self.zi_hdawg8.update_gain(0.5)
-        calls = [call.set('sigouts_{}_range'.format(ch), 0.5) for ch in range(8)]
+        calls = [call.set('sigouts_{}_range'.format(ch), 1.0) for ch in range(8)]
         self.awg.assert_has_calls(calls)
 
     def test_retrieve_gain(self):
         self.awg.get.return_value = 0.2
-        self.assertEqual(0.2, self.zi_hdawg8.retrieve_gain())
+        self.assertEqual(0.1, self.zi_hdawg8.retrieve_gain())
 
-        with self.assertRaises(ValueError):
+        with self.assertRaises(Exception):
```

Comment:
Here we're trying to get the ValueError raised in retrieve_gain when not all gains are the same. The reason you got an other exception than expected was because you now divide by 2 the value you got back from the lambda function that is defined here. So you get a string by int division error. So you are not testing what was meant in the old test.
When you change the lambda function for the side effect to:
self.awg.get.side_effect = lambda v: int(v[8:9])
we get the exception we want
```suggestion
        with self.assertRaises(ValueError):
```